src/recommenders/bo.py

```diff
@@ -1,19 +1,26 @@
+from abc import ABC, abstractmethod
 import pandas as pd
 from skopt import gp_minimize
 
 from src.configs import get_config
 from src.recommenders.base import BaseRecommender
 
 
-class BORecommender(BaseRecommender):
-    def recommend(self):
+class BORecommender(ABC):
+    @abstractmethod
+    def _BaseRecommender__recommend(self):
+        pass
+
+class Myrecommender(BaseRecommender):
+    __name__ = 'bayesian optimization'
+
+    def _BaseRecommender__recommend(self):
```

Comment:
Refactor. You don't need `BORecommender` as an ABC; `Myrecommender` should be called `BORecommender` instead. You can also now rename L17 to `_recommend`.